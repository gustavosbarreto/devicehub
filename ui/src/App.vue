<template>
<v-app>
    <v-navigation-drawer v-if="isLoggedIn" :clipped="false" fixed v-model="drawer" :mini-variant="false" enable-resize-watcher app>
        <v-container>
            <div class="text-center">
                <v-icon>mdi-console</v-icon>
                <h2 style="font-family: monospace">ShellHub</h2>
                <span class="overline">beta</span>
            </div>
        </v-container>
        <v-list>
            <v-list-item v-for="item in items" :key="item.title" :to="item.path" two-line>
                <v-list-item-action>
                    <v-icon v-text="item.icon"></v-icon>
                </v-list-item-action>

                <v-list-item-content>
                    <v-list-item-title v-text="item.title"></v-list-item-title>
                </v-list-item-content>
            </v-list-item>
        </v-list>
    </v-navigation-drawer>
    <v-app-bar app elevate-on-scroll color="primary" class="pl-3 pr-4" v-if="isLoggedIn">
        <v-menu transition="scale-transition" origin="top left">
            <template v-slot:activator="{ on }">
                <v-chip v-on="on">
                    <v-icon left>mdi-server</v-icon>
                    My Device Fleet
                    <v-icon right>mdi-chevron-down</v-icon>
                </v-chip>
            </template>
        </v-menu>
        <v-spacer></v-spacer>
        <v-chip>
            <v-icon>help</v-icon>
        </v-chip>
        <v-chip>
            <v-icon>notifications</v-icon>
        </v-chip>
        <v-menu transition="scale-transition" origin="top right">
            <template v-slot:activator="{ on }">
                <v-chip v-on="on">
                    <v-icon left>mdi-account</v-icon>
                    {{ $store.getters["auth/currentUser"] }}
                    <v-icon right>mdi-chevron-down</v-icon>
                </v-chip>
            </template>

            <v-card>
                <v-list-item three-line>
                    <v-list-item-content>
                        <v-list-item-title class="mb-1">Tenant ID</v-list-item-title>
                        <v-list-item-subtitle>
                            <v-chip>
                                <span>
                                    {{ tenant }}
                                </span>
                                <v-icon right v-clipboard="tenant" v-clipboard:success="() => { copySnack = true; }">mdi-content-copy</v-icon>
                            </v-chip>
                        </v-list-item-subtitle>
                    </v-list-item-content>
                </v-list-item>

                <v-card-actions>
                    <v-btn small text @click="logout()">
                        Logout
                    </v-btn>
                </v-card-actions>
            </v-card>
        </v-menu>
    </v-app-bar>
    <v-content>
        <v-container class="pa-8" fluid>
            <router-view></router-view>
        </v-container>
        <v-snackbar v-model="copySnack" :timeout=3000>Tenant ID copied to clipboard</v-snackbar>
    </v-content>
</v-app>
</template>

<script>
export default {
  name: "App",

  methods: {
    logout() {
      this.$store.dispatch("auth/logout").then(() => {
        this.$router.push("/login");
      });
    }
  },

  computed: {
    tenant() {
      return this.$store.getters["auth/tenant"];
    },

    isLoggedIn() {
      return this.$store.getters["auth/isLoggedIn"];
    }
  },

  data() {
    return {
      drawer: true,
      clipped: false,
      copySnack: false,
      items: [
        {
          icon: "dashboard",
          title: "Dashboard",
          path: "/"
        },
        {
          icon: "devices",
          title: "Devices",
          path: "/devices"
        },
        {
          icon: "history",
          title: "Sessions",
          path: "/sessions"
        }
      ]
    };
  }
};
</script>

<style>
.v-list-active {
  border-left: 4px solid var(--v-primary-base);
}

.text-shadow {
  text-shadow: #000 0 0 6px;
  color: transparent;
}
</style>
