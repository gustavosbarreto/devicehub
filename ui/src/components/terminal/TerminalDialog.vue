<template>
<v-dialog v-model="show" max-width="1024px">
    
  <template v-slot:activator="{ on }">
    <v-icon @click="open()">mdi-console</v-icon>
  </template>
  
  <v-card>
    <v-toolbar dark color="primary">
      <v-btn icon dark @click="close()">
        <v-icon>close</v-icon>
      </v-btn>
      <v-toolbar-title>Terminal</v-toolbar-title>
  
      <v-spacer></v-spacer>
    
    </v-toolbar>
    
    <v-card class="ma-0 pa-6" v-if="showLoginForm" outlined>      
      <v-form ref="form" v-model="valid" @submit.prevent="connect()" lazy-validation>
        <v-text-field label="Username" v-model="username" ref="username" autofocus :rules="[rules.required]" :validate-on-blur="true"></v-text-field>
        <v-text-field label="Password" type="password" v-model="passwd" :rules="[rules.required]" :validate-on-blur="true"></v-text-field>
        <v-btn type="submit" color="primary" class="mt-4" rounded>Connect</v-btn>
      </v-form>
    </v-card>
    
    <div ref="terminal"></div>
  </v-card>

</v-dialog>
</template>

<script>
import { Terminal } from 'xterm';
import * as fit from 'xterm/lib/addons/fit/fit';
import * as attach from 'xterm/lib/addons/attach/attach';
import 'xterm/dist/xterm.css';

Terminal.applyAddon(fit);
Terminal.applyAddon(attach);

export default {
  name: 'TerminalDialog',

  props: ['uid'],

  data() {
    return {
      username: '',
      passwd: '',
      showLoginForm: true,
      valid: true,
      rules: {
        required: value => !!value || 'Required'
      }
    };
  },

  watch: {
    show(value) {
      if (!value) {
        if (this.ws) this.ws.close();
        if (this.xterm) this.xterm.destroy();

        this.username = '';
        this.passwd = '';
        this.showLoginForm = true;
      } else {
        requestAnimationFrame(() => {
          this.$refs.username.focus();
        });
      }
    }
  },

  computed: {
    show: {
      get() {
        return this.$store.getters['modals/terminal'] === this.$props.uid;
      },

      set(value) {
        if (value) {
          this.$store.dispatch('modals/toggleTerminal', this.$props.uid);
        } else {
          this.$store.dispatch('modals/toggleTerminal', '');
        }
      }
    }
  },

  methods: {
    open() {
      this.xterm = new Terminal({
        cursorBlink: true,
        fontFamily: 'monospace'
      });

      this.$store.dispatch('modals/toggleTerminal', this.$props.uid);

      if (this.xterm.element) {
        this.xterm.reset();
      }
    },

    close() {
      this.$store.dispatch('modals/toggleTerminal', '');
    },

    connect() {
      var protocolConnectionURL = '';

      if (!this.$refs.form.validate(true)) {
        return;
      }

      this.showLoginForm = false;
      this.$nextTick(() => this.xterm.fit());

      if (!this.xterm.element) {
        this.xterm.open(this.$refs.terminal);
      }

      this.xterm.fit();
      this.xterm.focus();

      const params = Object.entries({
        user: `${this.username}@${this.$props.uid}`,
        passwd: encodeURIComponent(this.passwd),
        cols: this.xterm.cols,
        rows: this.xterm.rows
      })
        .map(([k, v]) => {
          return `${k}=${v}`;
        })
        .join('&');

      if(location.protocol == "http:"){
        protocolConnectionURL = 'ws'
      }else{
        protocolConnectionURL = 'wss'
      }

      this.ws = new WebSocket(protocolConnectionURL+`://${location.host}/ws/ssh?${params}`);

      this.ws.onopen = () => {
        this.xterm.attach(this.ws, true, true);
      };

      this.ws.onclose = () => {
        this.xterm.detach(this.ws);
      };
    },

    showCopySnack() {
      this.copySnack = true;
    }
  }
};
</script>
