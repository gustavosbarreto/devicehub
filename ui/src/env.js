export default {
  /* eslint-disable no-param-reassign */
  install(Vue) {
    const env = {
      isHosted: window.env.HOSTED === 'true',
    };

    Vue.env = env;
    Vue.prototype.$env = env;
  },
  /* eslint-enable no-param-reassign */
};
