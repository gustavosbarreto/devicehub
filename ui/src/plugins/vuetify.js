import Vue from 'vue';
import Vuetify from 'vuetify/lib';
import '@mdi/font/css/materialdesignicons.css';
import 'font-logos/assets/font-logos.css';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
import { fas } from '@fortawesome/free-solid-svg-icons';
import Clipboard from 'v-clipboard';
import Fragment from 'vue-fragment';

//import 'vuetify/dist/vuetify.min.css'
//import '../styles/variables.scss'

Vue.component('font-awesome-icon', FontAwesomeIcon); // Register component globally
library.add(fas); // Include needed icons.

Vue.use(Vuetify);
Vue.use(Clipboard);
Vue.use(Fragment.Plugin);

const DEVICE_ICONS = {
  alpine: 'fl-alpine',
  arch: 'fl-archlinux',
  centos: 'fl-centos',
  coreos: 'fl-coreos',
  debian: 'fl-debian',
  devuan: 'fl-devuan',
  elementary: 'fl-elementary',
  fedora: 'fl-fedora',
  freebsd: 'fl-freebsd',
  gentoo: 'fl-gentoo',
  linuxmint: 'fl-linuxmint',
  mageia: 'fl-mageia',
  manjaro: 'fl-manjaro',
  mandriva: 'fl-mandriva',
  nixos: 'fl-nixos',
  opensuse: 'fl-opensuse',
  rhel: 'fl-redhat',
  sabayon: 'fl-sabayon',
  slackware: 'fl-slackware',
  ubuntu: 'fl-ubuntu',
  raspbian: 'fl-raspberry-pi',
  'ubuntu-core': 'fl-ubuntu',
  void: 'fl-void',
};

export default new Vuetify({
  iconfont: 'md',
  icons:{
    values:DEVICE_ICONS
  },
  theme: {
    dark: false,
    themes: {
      light: {
        primary: '#364576',
        secondary: '#b0bec5',
        accent: '#8c9eff',
        error: '#b71c1c',
      },
    }
  }
});
