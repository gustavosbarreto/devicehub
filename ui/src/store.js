import Vue from 'vue';
import Vuex from 'vuex';
import stats from '@/modules/stats';
import sessions from '@/modules/sessions';
import auth from '@/modules/auth';
import devices from '@/modules/devices';
import modals from '@/modules/modals';
import firewallrules from '@/modules/firewall_rules';
import notifications from '@/modules/notifications';

Vue.use(Vuex);

export default new Vuex.Store({
  modules: {
    devices, modals, stats, sessions, auth, firewallrules, notifications,
  },
});
