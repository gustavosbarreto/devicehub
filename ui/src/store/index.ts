import { InjectionKey } from "vue";
import { createStore, Store, useStore as vuexUseStore } from "vuex";

import { auth, AuthState } from "./modules/auth";
import { apiKeys, ApiKeysState } from "./modules/api_keys";
import { layout, LayoutState } from "./modules/layout";
import { users, UsersState } from "./modules/users";
import { tags, TagsState } from "./modules/tags";
import { stats, StatsState } from "./modules/stats";
import { spinner, SpinnerState } from "./modules/spinner";
import { snackbar, SnackbarState } from "./modules/snackbar";
import { sessions, SessionsState } from "./modules/sessions";
import { security, SecurityState } from "./modules/security";
import { publicKeys, PublicKeysState } from "./modules/public_keys";
import { privateKey, PrivateKeyState } from "./modules/private_key";
import { notifications, NotificationsState } from "./modules/notifications";
import { modal, ModalState } from "./modules/modal";
import { firewallRules, FirewallRulesState } from "./modules/firewall_rules";
import { devices, DevicesState } from "./modules/devices";
import { box, BoxState } from "./modules/box";
import { namespaces, NamespacesState } from "./modules/namespaces";
import { billing } from "./modules/billing";
import { customer, CustomerState } from "./modules/customer";
import { announcement, AnnouncementState } from "./modules/announcement";
import apiPlugin from "./plugins/api";

export interface State {
  auth: AuthState;
  apiKeys: ApiKeysState;
  billing: NamespacesState;
  customer: CustomerState;
  box: BoxState;
  devices: DevicesState;
  firewallRules: FirewallRulesState;
  layout: LayoutState;
  modal: ModalState;
  namespaces: NamespacesState;
  notifications: NotificationsState;
  privateKey: PrivateKeyState;
  publicKeys: PublicKeysState;
  security: SecurityState;
  sessions: SessionsState;
  snackbar: SnackbarState;
  spinner: SpinnerState;
  stats: StatsState;
  tags: TagsState;
  users: UsersState;
  announcement: AnnouncementState;
}

export const key: InjectionKey<Store<State>> = Symbol("store");

export const store = createStore<State>({
  modules: {
    auth,
    apiKeys,
    customer,
    billing,
    box,
    devices,
    firewallRules,
    layout,
    modal,
    namespaces,
    notifications,
    privateKey,
    publicKeys,
    security,
    sessions,
    snackbar,
    spinner,
    stats,
    tags,
    users,
    announcement,
  },
  plugins: [
    apiPlugin,
  ],
});

export function useStore(): Store<State> {
  return vuexUseStore(key);
}
