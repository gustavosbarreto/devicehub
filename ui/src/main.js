import Vue from 'vue';

import Snackbar from '@/components/snackbar/Snackbar';

import App from './App';
import router from './router';
import store from './store';
import env from './env';
import './vee-validate';
import vuetify from './plugins/vuetify';

Vue.config.productionTip = false;

Vue.component('Snackbar', Snackbar);

Vue.use(require('vue-moment'));

Vue.use(env);

new Vue({
  vuetify,
  router,
  store,
  render: (h) => h(App),
}).$mount('#app');
