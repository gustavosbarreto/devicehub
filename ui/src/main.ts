import { createApp } from "vue";
import * as Sentry from "@sentry/vue";
import { BrowserTracing } from "@sentry/tracing";
import { envVariables } from "./envVariables";
import vuetify from "./plugins/vuetify";
import { key, store } from "./store";
import { router } from "./router";
import App from "./App.vue";

import { loadFonts } from "./plugins/webfontloader";

import SnackbarComponent from "./components/Snackbar/Snackbar.vue";
import { SnackbarPlugin } from "./plugins/snackbar";

const app = createApp(App);

Sentry.init({
  app,
  dsn: envVariables.sentryDsn || "",
  integrations: [
    new BrowserTracing({
      routingInstrumentation: Sentry.vueRouterInstrumentation(router),
    }),
  ],
  tracesSampleRate: 1.0,
  hooks: ["activate", "create", "destroy", "mount", "update"],
  timeout: 500,
  release: envVariables.version || "latest",
});
Sentry.setTag("project", "shellhub-ui");

loadFonts();

app.use(vuetify);
app.use(router);
app.use(store, key);
app.use(SnackbarPlugin);
app.component("SnackbarComponent", SnackbarComponent);
app.mount("#app");
