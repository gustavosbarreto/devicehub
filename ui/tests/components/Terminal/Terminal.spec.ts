import { createVuetify } from "vuetify";
import { flushPromises, mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it, vi } from "vitest";
import TerminalDialog from "../../../src/components/Terminal/TerminalDialog.vue";
import { createStore } from "vuex";
import { key } from "../../../src/store";
import routes from "../../../src/router";

const uid = "a582b47a";
const username = "user";
const password = "pass";
const tabActive = "Password";

const privateKeys = [
  {
    name: "shellhub",
    data: "BBGVvbmF",
  },
  {
    name: "shellhub",
    data: "AbGVvbmF",
  },
];

const store = createStore({
  state: {
    terminal: uid,
    privateKeys,
  },
  getters: {
    "modals/terminal": (state) => state.terminal,
    "privateKey/list": (state) => state.privateKeys,
  },
  actions: {
    "modals/toggleTerminal": vi.fn(),
  },
});

describe("TerminalDialog", () => {
  let wrapper: VueWrapper<any>;
  const vuetify = createVuetify();

  ///////
  // In this case, the rendering of the console icon is tested.
  // For this test to work, the uid in props is an empty string.
  ///////

  describe("Button", () => {
    beforeEach(() => {
      wrapper = mount(TerminalDialog, {
        global: {
          plugins: [[store, key], routes, vuetify],
        },
        props: {
          uid: "",
          show: false,
          enableConnectButton: true,
        },
      });
    });

    ///////
    // Component Rendering
    //////

    it("Is a Vue instance", () => {
      expect(wrapper).toBeTruthy();
    });
    it("Renders the component", () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data and Props checking
    //////

    it("Receive data in props", () => {
      expect(wrapper.vm.uid).toEqual("");
      expect(wrapper.vm.show).toEqual(false);
    });
    it("Compare data with default value", () => {
      expect(wrapper.vm.username).toEqual("");
      expect(wrapper.vm.password).toEqual("");
      expect(wrapper.vm.showLoginForm).toEqual(true);
      expect(wrapper.vm.tabActive).toEqual(tabActive);
    });
    it("Receive data in computed", () => {
      expect(wrapper.vm.showTerminal).toEqual(false);
      expect(wrapper.vm.show).toEqual(false);
      expect(wrapper.vm.getListPrivateKeys).toEqual(privateKeys);
    });
    it("Check the watch action", async () => {
      wrapper.vm.username = username;
      wrapper.vm.password = password;

      expect(wrapper.vm.username).toEqual(username);
      expect(wrapper.vm.password).toEqual(password);
    });

    //////
    // HTML validation
    //////

    it("Renders the template with data", () => {
      expect(wrapper.find('[data-test="connect-btn"]').exists()).toBe(true);
      expect(wrapper.find('[data-test="console-icon"]').exists()).toBe(false);
      expect(wrapper.find('[data-test="console-item"]').exists()).toBe(false);
      expect(wrapper.find('[data-test="terminal-dialog"]').exists()).toBe(
        false
      );
    });
  });

  describe("Icon", () => {
    beforeEach(() => {
      wrapper = mount(TerminalDialog, {
        global: {
          plugins: [[store, key], routes, vuetify],
        },
        props: {
          uid: "",
          show: false,
        },
      });
    });

    ///////
    // Component Rendering
    //////

    it("Is a Vue instance", () => {
      expect(wrapper).toBeTruthy();
    });
    it("Renders the component", () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data and Props checking
    //////

    it("Receive data in props", () => {
      expect(wrapper.vm.uid).toEqual("");
      expect(wrapper.vm.show).toEqual(false);
    });

    it("Compare data with default value", () => {
      expect(wrapper.vm.username).toEqual("");
      expect(wrapper.vm.password).toEqual("");
      expect(wrapper.vm.showLoginForm).toEqual(true);
      expect(wrapper.vm.tabActive).toEqual(tabActive);
    });
    it("Receive data in computed", () => {
      expect(wrapper.vm.showTerminal).toEqual(false);
      expect(wrapper.vm.show).toEqual(false);
      expect(wrapper.vm.getListPrivateKeys).toEqual(privateKeys);
    });
    it("Check the watch action", async () => {
      wrapper.vm.username = username;
      wrapper.vm.password = password;

      expect(wrapper.vm.username).toEqual(username);
      expect(wrapper.vm.password).toEqual(password);
    });

    //////
    // HTML validation
    //////

    it("Renders the template with data", () => {
      expect(wrapper.find('[data-test="connect-btn"]').exists()).toBe(false);
      expect(wrapper.find('[data-test="console-icon"]').exists()).toBe(true);
      expect(wrapper.find('[data-test="console-item"]').exists()).toBe(true);
      expect(wrapper.find('[data-test="terminal-dialog"]').exists()).toBe(
        false
      );
    });
  });

  ///////
  // In this case the dialog is opened
  ///////

  describe("Dialog opened", () => {
    beforeEach(() => {
      wrapper = mount(TerminalDialog, {
        global: {
          plugins: [[store, key], routes, vuetify],
        },
        props: {
          uid,
          show: true,
          enableConnectButton: true,
        },
        data() {
          return {
            showTerminal: false,
          }
        }
      });
    });

    ///////
    // Component Rendering
    //////

    it("Is a Vue instance", () => {
      expect(wrapper).toBeTruthy();
    });
    it("Renders the component", () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data and Props checking
    //////

    it("Receive data in props", () => {
      expect(wrapper.vm.uid).toEqual(uid);
      expect(wrapper.vm.show).toEqual(true);
      expect(wrapper.vm.enableConnectButton).toEqual(true);
    });

    it("Compare data with default value", () => {
      expect(wrapper.vm.username).toEqual("");
      expect(wrapper.vm.password).toEqual("");
      expect(wrapper.vm.showLoginForm).toEqual(true);
      expect(wrapper.vm.tabActive).toEqual(tabActive);
    });
    it("Receive data in computed", () => {
      expect(wrapper.vm.showTerminal).toEqual(false);
      expect(wrapper.vm.show).toEqual(true);
      expect(wrapper.vm.getListPrivateKeys).toEqual(privateKeys);
    });
    it("Check the watch action", async () => {
      wrapper.vm.username = username;
      wrapper.vm.password = password;

      expect(wrapper.vm.username).toEqual(username);
      expect(wrapper.vm.password).toEqual(password);
    });

    //////
    // HTML validation
    //////

    it("Renders the template with data", () => {
      // TODO
      expect(wrapper.find('[data-test="connect-btn"]').exists()).toBe(true);
      expect(wrapper.find('[data-test="console-icon"]').exists()).toBe(false);
      expect(wrapper.find('[data-test="console-item"]').exists()).toBe(false);
    });
  });
});
