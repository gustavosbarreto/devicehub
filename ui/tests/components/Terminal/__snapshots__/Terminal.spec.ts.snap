// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TerminalDialog > Button > Renders the component 1`] = `
"<button type=\\"button\\" class=\\"v-btn v-btn--disabled v-theme--light text-normal v-btn--density-comfortable v-btn--size-default v-btn--variant-outlined\\" disabled=\\"\\" data-test=\\"connect-btn\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
  <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\">Offline</span>
  <!---->
  <!---->
</button>
<!--teleport start-->
<!--teleport end-->"
`;

exports[`TerminalDialog > Dialog opened > Renders the component 1`] = `
"<button type=\\"button\\" class=\\"v-btn v-btn--disabled v-theme--light text-normal v-btn--density-comfortable v-btn--size-default v-btn--variant-outlined\\" disabled=\\"\\" data-test=\\"connect-btn\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
  <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\">Offline</span>
  <!---->
  <!---->
</button>
<!--teleport start-->
<!--teleport end-->"
`;

exports[`TerminalDialog > Icon > Renders the component 1`] = `
"<span><i class=\\"mdi-console mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\" left=\\"\\" data-test=\\"console-icon\\"></i></span>
<span><div class=\\"v-list-item-title ml-2\\" data-test=\\"console-item\\"> Console </div></span>
<!--teleport start-->
<!--teleport end-->"
`;
